src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorSpan.cs

```diff
@@ -132,7 +132,7 @@ public TensorSpan(Span<T> span, scoped ReadOnlySpan<nint> lengths, scoped ReadOn
             TensorSpanHelpers.ValidateStrides(strides, lengths);
 
             nint maxElements = TensorSpanHelpers.ComputeMaxLinearIndex(strides, lengths);
-            if (maxElements >= span.Length && span.Length != 0)
+            if ((maxElements >= span.Length && span.Length != 0) || (span.Length == 0 && maxElements > 0))
```

Comment:
Why the different checks in `TensorSpan` vs `ReadOnlyTensorSpan`?

Reply:
Fixed.